eth2/state_processing/src/per_block_processing/verify_deposit.rs

```diff
@@ -10,19 +11,22 @@ pub fn verify_deposit_signature<T: EthSpec>(
     state: &BeaconState<T>,
     deposit: &Deposit,
     spec: &ChainSpec,
+    pubkey: &PublicKey,
 ) -> Result<(), Error> {
     // Note: Deposits are valid across forks, thus the deposit domain is computed
     // with the fork zeroed.
     let domain = spec.get_domain(state.current_epoch(), Domain::Deposit, &Fork::default());
-    verify!(
-        deposit
-            .data
-            .signature
-            .verify(&deposit.data.signed_root(), domain, &deposit.data.pubkey,),
-        Invalid::BadSignature
-    );
-
-    Ok(())
+    let parsed_signature: Result<Signature, _> = (&deposit.data.signature).try_into();
+    match parsed_signature {
+        Err(_) => Err(Error::Invalid(Invalid::BadSignatureBytes)),
+        Ok(signature) => {
+            verify!(
+                signature.verify(&deposit.data.signed_root(), domain, pubkey,),
+                Invalid::BadSignature
+            );
+            Ok(())
+        }
+    }
```

Comment:
Just a minor style thing, but this could be written:

```rust
    let signature: Signature = (&deposit.data.signature)
        .try_into()
        .map_err(|_| Error::Invalid(Invalid::BadSignatureBytes))?;

    verify!(
        signature.verify(&deposit.data.signed_root(), domain, pubkey),
        Invalid::BadSignature
    );

    Ok(())
```

Which is a bit clearer IMO.